dartagnan/src/main/java/com/dat3m/dartagnan/program/processing/Intrinsics.java

```diff
@@ -1328,6 +1331,106 @@ private List<Event> inlineMemCpy(FunctionCall call) {
         return replacement;
     }
 
+    // https://en.cppreference.com/w/c/string/byte/memcpy
+    private List<Event> inlineMemCpyS(FunctionCall call) {
+        // Cast guaranteed to success by the return type of memcpy_s
+        final Register resultRegister = ((ValueFunctionCall)call).getResultRegister();
+        final Function caller = call.getFunction();
+        final Expression dest = call.getArguments().get(0);
+        // This parameter has type rsize_t/size_t which we model as types.getArchType(), thus the cast
+        final Expression destszExpr = expressions.makeCast(call.getArguments().get(1), types.getArchType());
+        final Expression src = call.getArguments().get(2);
+        // This parameter has type rsize_t/size_t which we model as types.getArchType(), thus the cast
+        final Expression countExpr = expressions.makeCast(call.getArguments().get(3), types.getArchType());
+
+        if (!(countExpr instanceof IntLiteral countValue)) {
+            final String error = "Cannot handle memcpy_s with dynamic count argument: " + call;
+            throw new UnsupportedOperationException(error);
+        }
```

Comment:
This is not going to work. You generate the cast expression and then check for it to be a literal, which should always fail. The reason it works right now is that the cast is recognized as NOOP and optimized away immediately.